File: Medium/C#/PowerSet.cs
using System;
using System.Collections.Generic;

public class Powerset
{
    public static List<List<int>> Powersets(List<int> array)
    {
        return GetPowerset(array, 0);
    }

    private static List<List<int>> GetPowerset(List<int> array, int index)
    {
        if (index == array.Count)
        {
            return new List<List<int>> { new List<int>() };
        }

        int currentElement = array[index];
        List<List<int>> subsets = GetPowerset(array, index + 1);
        int length = subsets.Count;

        for (int i = 0; i < length; i++)
        {
            List<int> currentSubset = new List<int>(subsets[i]);
            currentSubset.Add(currentElement);
            subsets.Add(currentSubset);
        }

        return subsets;
    }

    public static void Main(string[] args)
    {
        List<int> array = new List<int> { 1, 2, 3 };
        List<List<int>> result = Powersets(array);

        Console.WriteLine("Powerset: ");
        foreach (List<int> subset in result)
        {
            Console.Write("[");
            foreach (int element in subset)
            {
                Console.Write($"{element}, ");
            }
            Console.WriteLine("]");
        }
    }
}

File: Medium/JAVA/PowerSet.java
import java.util.ArrayList;
import java.util.List;

public class PowerSet {
    public static List<List<Integer>> powerset(List<Integer> array) {
        return getPowerset(array, 0);
    }

    private static List<List<Integer>> getPowerset(List<Integer> array, int index) {
        if (index == array.size()) {
            List<List<Integer>> emptySet = new ArrayList<>();
            emptySet.add(new ArrayList<>());
            return emptySet;
        }

        int currentElement = array.get(index);
        List<List<Integer>> subsets = getPowerset(array, index + 1);
        int length = subsets.size();

        for (int i = 0; i < length; i++) {
            List<Integer> currentSubset = new ArrayList<>(subsets.get(i));
            currentSubset.add(currentElement);
            subsets.add(currentSubset);
        }

        return subsets;
    }

    public static void main(String[] args) {
        List<Integer> array = new ArrayList<>();
        array.add(1);
        array.add(2);
        array.add(3);
        List<List<Integer>> result = powerset(array);

        System.out.println("Powerset: ");
        for (List<Integer> subset : result) {
            System.out.print("[");
            for (int element : subset) {
                System.out.print(element + ", ");
            }
            System.out.println("]");
        }
    }
}
